clients/geth/specular/rollup/services/validator/validator.go

```diff
@@ -59,18 +59,36 @@ func (v *Validator) tryValidateAssertion(lastValidatedAssertion, assertion *roll
 	currentBlockNum := assertion.StartBlock
 	currentChainHeight := v.Chain().CurrentBlock().NumberU64()
 	var block *types.Block
+
+	log.Trace(
+		"Trying to validate assertion",
+		"#currentBlockNum", currentBlockNum,
+		"#currentChainHeight", currentChainHeight,
+		"#inboxSizeDiff", inboxSizeDiff.Uint64(),
+	)
+
 	for inboxSizeDiff.Cmp(common.Big0) > 0 {
+		log.Trace(
+			"Looping through inbox",
+			"#inboxSizeDiff", inboxSizeDiff.Uint64(),
+			"#currentBlockNum", currentBlockNum,
+		)
+
 		if currentBlockNum > currentChainHeight {
 			return errAssertionOverflowedLocalInbox
 		}
+
 		block = v.Chain().GetBlockByNumber(currentBlockNum)
+
 		if block == nil {
+			log.Warn("Could not get current block by number")
```

Comment:
I'd add this to the error msg directly by making `errAssertionOverflowedLocalInbox` into a struct and adding a Msg field which appends to the existing message in `Error()`